site/src/components/MdxPageContent.tsx

```diff
@@ -22,50 +22,71 @@ export const MdxPageContent: VFC<MdxPageContentProps> = ({
 
   const [headings, setHeadings] = useState<Heading[]>([]);
 
-  useEffect(() => {
-    setHeadings([]);
-  }, [serializedPage]);
-
-  const [currentHeading, setCurrentHeading] = useState<Heading>();
   const [detectHeadingFromScroll, setDetectHeadingFromScroll] =
     useState<boolean>(true);
 
-  const [anchor, setAnchor] = useState<string>(
-    router.asPath.split("#")[1] ?? "",
-  );
+  const currentHeading = useRef<Heading | undefined>(undefined);
```

Comment:
Converting `currentHeading` into a ref and removing the state hook for `anchor` helped as well, since they were causing more effect triggers than needed when updated.